src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -599,6 +608,18 @@ void RunCompaction()
                                 this.Log.BeginAddress - beginAddressBeforeCompaction,
                                 this.GetElapsedCompactionMilliseconds());
 
+
+                            long mutableSizeAfterCompaction = this.Log.TailAddress - this.Log.SafeReadOnlyAddress;
+
+                            if (this.Log.SafeReadOnlyAddress == compactedUntil && mutableSizeBeforeCompaction == 0 && readOnlySizeBeforeCompaction == mutableSizeAfterCompaction)
```

Comment:
I think this is a good opportunity to explicit document the relationship between the read-only and mutable segments with compaction. Can we write a few comments about how we expect them to change with compaction (plus a refresher on how data enters and leaves the read-only segment). I think that's important to understand why this detection works.

Reply:
I have added extensive comments to explain compaction.

I also made turned this from a simple trace statement into a warning. This will help us assess the priority of how often this happens and serve as a reminder.